Chess/src/Bishop.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Bishop.h"
+
+// It moves and captures along diagonals 
+vector<pair<int, int>> Bishop::getBasicMoves()
+{
+	return Bishop::getValidMoves();
```

Comment:
there's no need to qualify the function call to getValidMoves() with Bishop::. Since you're already inside the Bishop class, you can directly call getValidMoves().